autogen/agentchat/conversable_agent.py

```diff
@@ -708,6 +721,7 @@ def initiate_chat(
         for agent in [self, recipient]:
             agent.client_cache = agent.previous_cache
             agent.previous_cache = None
+        return self._get_chat_takeaway(context.get("takeaway_method"), recipient, prompt=context.get("takeaway_prompt"))
```

Comment:
I have been thinking about adding `ChatResult` as well. Just not sure if it may delay the merge of this PR. @sonichi needs your input. Should we add it? If we should, shall we do this in this PR or a later one?

Reply:
I'm convinced that a Chat class is OK to add. We already have a GroupChat class.
initiate_chat can return a Chat object
initiate_chats can return a list of Chat
Let's do it in a separate PR.